### DISCUSSION TITLE
Genymotion PaaS supported?

### DISCUSSION THREAD

**Discussion by adrienharnay** [wix/Detox#3913] [Q&A]
Hello,

I would like to integrate Genymotion via AWS (EC2 instances). Does Detox support their "Device image" offer?

Thanks in advance and have a nice day!

**Comment by d4vidi** [comment]
Hey @adrienharnay!
My familiarity with Genymobile's PaaS is limited, but I will try to provide an answer nonetheless (my best one). The Detox<>Genymotion integration comes into effect on two levels:
1. Cloud instances allocation and tear-down.
2. Interaction with the instances via ADB.

While the 1st level deeply relies on Genymobile's [`gmsaas`](https://docs.genymotion.com/gmsaas) - which is _not_ suitable for PaaS, you could probably work around it (i.e. by asking Detox to manage the cloud-instances on your own). The 2nd level, however, requires that the cloud instance would be visible via `ADB`. With `gmsaas`, you can achieve that using the simple `gmsaas instances adbconnect` command. However, I'm inclined to say that in a PaaS environment, that is unlikely to be possible.

@lilinor perhaps you would have more insights in regards.

**Reply by lilinor** [reply]
Hey @adrienharnay !

@d4vidi is right, the current integration is only compatible with Genymotion SaaS but you can definitely use Genymotion Device Image on AWS with detox. You will need to manage the ec2 instances on your own through aws ec2 apis and connect to the devices with adb using the `adb connect` command. You can refer to our documentation on how to do that [here](https://docs.genymotion.com/paas/Access/04_ADB/)
In your detox config file you will use the `android.attached` type. 

Don't hesitate to contact me directly by email (ekwok@genymobile.com) so we can discuss further on your requirements, technical questions you might have and why not in the future about an integration with Genymotion PaaS & detox :)

**Reply by adrienharnay** [reply]
Thank you both for your answers!
